### PR TITLE
Issue #365 Add resilience to job manager

### PR DESCRIPTION
**Pull request by JohanKJSchreurs**
Implements issue https://github.com/Open-EO/openeo-python-client/issues/365

### CODE REVIEW

File: openeo/extra/job_management.py [hunk 1]
```diff
@@ -95,6 +101,57 @@ def add_backend(
...
+        retries = Retry(
+            total=5,
+            read=50,
+            other=50,
+            status=50,
+            backoff_factor=0.1,
+            status_forcelist=status_forcelist,
+            allowed_methods=["HEAD", "GET", "OPTIONS", "POST"],
+        )
+        connection.session.mount("https://", HTTPAdapter(max_retries=retries))
```
Comment (jdries) [inline openeo/extra/job_management.py:152]: Here we rely on the fact that our Connection implementation is based on requests, but not sure how to improve this. I would perhaps do it like this for now.

File: openeo/extra/job_management.py [hunk 2]
```diff
@@ -121,17 +178,40 @@ def _normalize_df(self, df: pd.DataFrame) -> pd.DataFrame:
...
+                    Like connection in add_backend:
+                    - either a Connection to the backend,
+                    - or a callable to create a backend connection.
+
+                connection:
+                    The Connection itself, that has already been created.
+
+                provider:
+                    The name of the backend that will run the job.
+
```
Comment (JohanKJSchreurs) [inline openeo/extra/job_management.py:214]: 1) The way we define what start_job should look like is a bit awkward. I think start_job would be a good candidate for using a Protocol, that defines and documents what the expected / supported function's parameters are, but Protocol was only introduced in Python 3.8.
Maybe a TODO for when we drop support for python 3.7 and 3.6?

2) That said we need to document what parameters the job manager passes to start_jobs so we know what parameters you can (maximally) have in your implementation.
This way to document it is a little clunky but I think enough for now.
Suggestions for a better solution are welcome.

3) We allow two similar parameters which is a little confusion: both a Connection as well as a callback to create a connection.
But I think the reason was that in practice the implementation would use either one or the other, if it is using them at all. If this is correct, that can help improve to improve the docstring.

Also, it looks like the intention is that start_job always uses keyword arguments and `*args, **kwargs`, so you can simply leave out parameters that you aren't going to use, even though the job manager will pass them to start_jobs.

Comment (JohanKJSchreurs) [reply openeo/extra/job_management.py]: I updated the docstring to make it more clear about the points above.
That is probably good enough for now.

**Comment by jdries** [comment]
looks good, merging!

**Pull request merged**